core/frontend/src/components/version-chooser/VersionChooser.vue

```diff
@@ -0,0 +1,506 @@
+      />
+      <v-progress-linear
+        v-if="disable_upload_controls && upload_percentage !== 100"
+        class="mb-4"
+        :value="upload_percentage"
+      />
+      <v-btn
+        :color="'primary'"
+        class="mr-2 mb-4"
+        @click="upload()"
+        v-text="'Upload'"
+      />
+    </v-card>
+    <v-overlay
+      :z-index="10"
+      :value="waiting"
+    >
+      <spinning-logo
+        size="100%"
+      />
+    </v-overlay>
+  </div>
+</template>
+
+<script lang="ts">
+import { gt, SemVer } from 'semver'
+import Vue from 'vue'
+
+import settings from '@/libs/settings'
+import { Dictionary } from '@/types/common'
+import { Version, VersionsQuery } from '@/types/version-chooser'
+import back_axios from '@/utils/api'
+
+import SpinningLogo from '../common/SpinningLogo.vue'
+import VersionCard from './VersionCard.vue'
+
+export default Vue.extend({
+  name: 'VersionChooser',
+  components: {
+    SpinningLogo,
+    VersionCard,
+  },
+  data() {
+    return {
+      settings,
+      pull_output: '',
+      show_pull_output: false,
+      available_versions: {
+        local: [],
+        remote: [],
+        error: null,
+      } as VersionsQuery,
+      latest_stable: undefined as (undefined | string),
+      latest_beta: undefined as (undefined | string),
+      disable_upload_controls: false,
+      upload_percentage: 0,
+      current_version: null as (null | Version),
+      loading_images: false,
+      waiting: false,
+      selected_image: 'bluerobotics/companion-core',
+      deleting: '', // image currently being deleted, if any
+      error: null as (null | string),
+    }
+  },
+  computed: {
+  },
+  mounted() {
+    this.loadCurrentVersion()
+  },
+  methods: {
+    async checkIfBackendIsOnline() {
+      await back_axios({
+        method: 'get',
+        url: '/version-chooser/v1.0/version/current',
+        timeout: 500,
+      })
+        .then(() => {
+          if (this.waiting) {
+            // Allow 3 seconds so the user can read the "complete" message
+            // reload(true) forces the browser to fetch the page again
+            setTimeout(() => { window.location.reload(true) }, 3000)
+          }
+        })
+        .catch((error) => {
+          console.log(error)
+          this.waiting = true
+        })
+    },
+    fixVersion(version: string) {
+      /** It turned out that our semvers are wrong... oopss
+      This turns 1.0.0.beta12 into 1.0.0-beta.12
+      Additionally filters out tags with no '.' in it, which
+      can be improved
+      */
+      if (version.includes('beta') && !version.includes('-beta.')) {
+        return version.replace('.beta', '-beta.')
+      }
+      if (!version.includes('.')) {
+        return null
+      }
+      return version
+    },
+    isSemVer(version: string): boolean {
+      // validates a version as SemVer compliant
+      try {
+        const fixed_version = this.fixVersion(version)
+        if (fixed_version == null) {
```

Comment:
This check is unnecessary from what I can understand reading the code, the `new SemVer` should raise an error and return false

Reply:
typescript won't allow 'null' as a paraeter for SemVer(), that's why the check is there